core/server/src/main/java/alluxio/worker/file/FileDataManager.java

```diff
@@ -302,42 +301,13 @@ public void persistFile(long fileId, List<Long> blockIds) throws AlluxioExceptio
    * @throws IOException if the folder creation fails
    */
   private String prepareUfsFilePath(long fileId) throws AlluxioException, IOException {
-    String ufsRoot = Configuration.get(PropertyKey.UNDERFS_ADDRESS);
     FileInfo fileInfo = mBlockWorker.getFileInfo(fileId);
-    AlluxioURI uri = new AlluxioURI(fileInfo.getPath());
-    AlluxioURI dstPath = new AlluxioURI(PathUtils.concatPath(ufsRoot, fileInfo.getPath()));
-    LOG.info("persist file {} at {}", fileId, dstPath);
-    String parentPath = PathUtils.concatPath(ufsRoot, uri.getParent().getPath());
-    // creates the parent folder if it does not exist
-    if (!mUfs.isDirectory(parentPath)) {
-      FileSystem fs = FileSystem.Factory.get();
-      // Create ancestor directories from top to the bottom. We cannot use recursive create parents
-      // here because the permission for the ancestors can be different.
-      Stack<Pair<String, MkdirsOptions>> ufsDirsToMakeWithOptions = new Stack<>();
-      AlluxioURI curAlluxioPath = uri.getParent();
-      AlluxioURI curUfsPath = dstPath.getParent();
-      // Stop at the Alluxio root because the mapped directory of Alluxio root in UFS may not exist.
-      while (!mUfs.isDirectory(curUfsPath.toString()) && curAlluxioPath != null) {
-        URIStatus curDirStatus;
-        curDirStatus = fs.getStatus(curAlluxioPath);
-        Permission perm = new Permission(curDirStatus.getOwner(), curDirStatus.getGroup(),
-            (short) curDirStatus.getMode());
-        ufsDirsToMakeWithOptions.push(new Pair<>(curUfsPath.toString(),
-            MkdirsOptions.defaults().setCreateParent(false).setPermission(perm)));
-        curAlluxioPath = curAlluxioPath.getParent();
-        curUfsPath = curUfsPath.getParent();
-      }
-      while (!ufsDirsToMakeWithOptions.empty()) {
-        Pair<String, MkdirsOptions> ufsDirAndPerm = ufsDirsToMakeWithOptions.pop();
-        // UFS mkdirs might fail if the directory is already created. If so, skip the mkdirs
-        // and assume the directory is already prepared, regardless of permission matching.
-        if (!mUfs.mkdirs(ufsDirAndPerm.getFirst(), ufsDirAndPerm.getSecond())
-            && !mUfs.isDirectory(ufsDirAndPerm.getFirst())) {
-          throw new IOException("Failed to create dir: " + ufsDirAndPerm.getFirst());
-        }
-      }
-    }
-    return dstPath.toString();
+    AlluxioURI alluxioPath = new AlluxioURI(fileInfo.getPath());
+    FileSystem fs = FileSystem.Factory.get();
```

Comment:
thanks for fixing this -- we should not have assumed `UNDERFS_ADDRESS` is always the correct ufs prefix.